### DISCUSSION TITLE
changing iconPath in treeView

### DISCUSSION THREAD

**Discussion by DmitryOffsec** [microsoft/vscode-discussions#881] [Extension Development QnA] [answered]
I have a TreeView with various elements. I want to change the icon of a tree element, which displays its status. During the execution of the command, I change the IconPath to the desired one, but it is not updated in the tree. 
How can I change the iconPath of a single node?
Thanks in advance!

**Comment by tamuratak** [comment, marked as answer]
You have to call `treeDataProvider.onDidChangeTreeData.fire(element)` to tell the VS Code runtime that the tree item has changed. See https://code.visualstudio.com/api/references/vscode-api#TreeDataProvider.onDidChangeTreeData

The following is a sample:

- https://github.com/microsoft/vscode-extension-samples/blob/a8a74c4d7773470e3d53cb89502b442579d5de84/tree-view-sample/src/jsonOutline.ts#L5-L33

**Reply by DmitryOffsec** [reply]
Thank you very much this works great!
